// sequelize-cli/bin/migrations/20141124102157-init-db.js
/**
 * Initial database migration for JanusVR server
 *
 * Adds database support for user registration
 *
 * @type {{up: Function, down: Function}}
 */



module.exports = {
    up: function(migration, DataTypes, done) {
        // add altering commands here, calling 'done' when finished

        migration.createTable(
            'users',
            {
                id: {
                    type: DataTypes.INTEGER,
                    primaryKey: true,
                    autoIncrement: true
                },
                createdAt: {
                    type: DataTypes.DATE
                },
                updatedAt: {
                    type: DataTypes.DATE
                },
                firstName: DataTypes.STRING,
                lastName: DataTypes.STRING,
                username: DataTypes.STRING,
                password: DataTypes.STRING,
                email: DataTypes.STRING,
                ip: DataTypes.STRING,
                enabled: {
                    type: DataTypes.BOOLEAN,
                    defaultValue: false,
                    allowNull: false
                }
            },
            {
                engine: 'InnoDB', // default: 'InnoDB'
                charset: 'utf8' // default: null
            }
        )

        migration.createTable(
            'groups',
            {
                id: {
                    type: DataTypes.INTEGER,
                    primaryKey: true,
                    autoIncrement: true
                },
                createdAt: {
                    type: DataTypes.DATE
                },
                updatedAt: {
                    type: DataTypes.DATE
                },

                name: DataTypes.STRING,
                description: DataTypes.STRING,

                enabled: {
                    type: DataTypes.BOOLEAN,
                    defaultValue: false,
                    allowNull: false
                },

                level: DataTypes.INTEGER
            },
            {
                engine: 'InnoDB', // default: 'InnoDB'
                charset: 'utf8' // default: null
            }
        )

        migration.createTable(
            'groupsUsers',
            {
                id: {
                    type: DataTypes.INTEGER,
                    primaryKey: true,
                    autoIncrement: true
                },
                createdAt: {
                    type: DataTypes.DATE
                },
                updatedAt: {
                    type: DataTypes.DATE
                },

                userId: DataTypes.INTEGER,
                groupId: DataTypes.INTEGER
            },
            {
                engine: 'InnoDB', // default: 'InnoDB'
                charset: 'utf8' // default: null
            }
        )

        migration.createTable(
            'groupsPermissions',
            {
                id: {
                    type: DataTypes.INTEGER,
                    primaryKey: true,
                    autoIncrement: true
                },
                createdAt: {
                    type: DataTypes.DATE
                },
                updatedAt: {
                    type: DataTypes.DATE
                },

                groupId: DataTypes.INTEGER,
                permissionId: DataTypes.INTEGER
            },
            {
                engine: 'InnoDB', // default: 'InnoDB'
                charset: 'utf8' // default: null
            }
        )

        migration.createTable(
            'permissions',
            {
                id: {
                    type: DataTypes.INTEGER,
                    primaryKey: true,
                    autoIncrement: true
                },
                createdAt: {
                    type: DataTypes.DATE
                },
                updatedAt: {
                    type: DataTypes.DATE
                },

                name: DataTypes.STRING,
                description: DataTypes.STRING
            },
            {
                engine: 'InnoDB', // default: 'InnoDB'
                charset: 'utf8' // default: null
            }
        )



        //Auth.register

        done()
    },
    down: function(migration, DataTypes, done) {
        // add reverting commands here, calling 'done' when finished
        migration.dropTable('users');
        migration.dropTable('groups');
        migration.dropTable('groupsUsers');
        done()
    }
}
