// sequelize-cli/bin/migrations/20141124102157-init-db.js
/**
 * Initial database migration for JanusVR server
 *
 * Adds database support for user registration
 *
 * @type {{up: Function, down: Function}}
 */

module.exports = {
  up: function(migration, DataTypes, done) {
    // add altering commands here, calling 'done' when finished

      migration.createTable(
          'users',
          {
              id: {
                  type: DataTypes.INTEGER,
                  primaryKey: true,
                  autoIncrement: true
              },
              createdAt: {
                  type: DataTypes.DATE
              },
              updatedAt: {
                  type: DataTypes.DATE
              },
              firstName: DataTypes.STRING,
              lastName: DataTypes.STRING,
              username: DataTypes.STRING,
              email: DataTypes.STRING,
              ip: DataTypes.STRING,
              groupId: DataTypes.INTEGER,
              enabled: {
                  type: DataTypes.BOOLEAN,
                  defaultValue: false,
                  allowNull: false
              }
          },
          {
              engine: 'InnoDB', // default: 'InnoDB'
              charset: 'utf8' // default: null
          }
      )

    done()
  },
  down: function(migration, DataTypes, done) {
    // add reverting commands here, calling 'done' when finished
      migration.dropAllTables().complete(function(err) {
          if (err) {
              done(err)
          } else {
              done(null)
          }
      })
  }
}
